consensus/reactor.go

```diff
@@ -745,10 +745,12 @@ OUTER_LOOP:
 			// which contains precommit signatures for prs.Height.
 			var ec *types.ExtendedCommit
 			var veEnabled bool
+			conR.conS.mtx.RLock()
```

Comment:
Should we surround the sleep with a lock?

Reply:
Sorry, I'm not done with this yet (my bad I marked it ready for review), there are still two commits missing (and some description missing as well)

Reply:
My bad, sorry! :D

Reply:
> My bad, sorry! :D

I disagree, it was my fault since I didn't think of working on a draft PR first. Sorry again